clowder-framework/clowder2-frontend
# Frontend crud files (without versions) #8

@max-zilla @tcnichol @lmarini also feel free to try below steps

How to test
- Make sure you build the latest fastapi main branch following the [README](https://github.com/clowder-framework/clowder2-fastapi/blob/main/README.md) and make sure you docker compose the latest build. I have to run `docker-compose up --scale backend=4 --force-recreate` on my machine
- Then run `export CLOWDER_REMOTE_HOSTNAME="http://clowder.docker.localhost"` and `npm start`
- Note: no need to run `npm run codegen:v2` unless you want to. We are committing all the clients in this PR
- You should be able to login/logout, create dataset, delete dataset, add files to dataset, download files, delete files

-------------------

Things happening in this PR:
- replace "fetch" with API client automatically generated with codegen IF it's available. This means this repo will stop working with V1
- add logout. You can either logout by clicking the logout link on the topbar; or directly navigate to `/logout` in the URL
- move create dataset and upload file from `/util` to `/action` and connect them with Redux. In V1 we were not able to get the created dataset/file object from API response hence we were not able to use Redux; now it's available
- disabled all the links/buttons that currently do not have backend support
